lib/finders/ComponentFinder.js

```diff
@@ -127,15 +127,26 @@ class ComponentFinder extends events.EventEmitter {
 
 	/**
 	 * Watches the components for changing.
+	 *
+	 * @return {Promise} Promise resolve when all watcher will be ready
 	 */
 	watch() {
-		if (this._fileWatcher) {
-			return;
+		if (this._readyWatchers) {
+			return this._readyWatchers;
 		}
 
-		this._fileWatcher = chokidar.watch(
-			Object.keys(this._foundComponentsByDirs), CHOKIDAR_OPTIONS
-		)
+		const readyDefers = [Promise.defer(), Promise.defer()];
+
+		this._readyWatchers = Promise
+			.all([readyDefers[0].promise, readyDefers[1].promise])
+			.then(data => {
+				this.emit('watchReady');
+				return data;
+			});
```

Comment:
Please avoid using `defer`. It's not a part of Promise spec and we can easily use `new Promise((resolve, reject) => ...)` here.